### PR TITLE
chore(Babel): Add emotion babel plugin to Core + Labs WEB-984

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview

Adds baseline source mapping and labelling.  We should move this to our own distributed babel presets when possible.

### PR Checklist

- [ ] Related to Abstract designs:
- [x] Related to JIRA ticket: WEB-984
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.1-alpha.3c8d5d.0<br/>@codecademy/gamut@12.3.0-alpha.3c8d5d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1f4105c07075b38c206e4f--gamut-preview.netlify.app>https://5f1f4105c07075b38c206e4f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f1f4105c07075b38c206e4f>Deploy Logs</a></p>

File: packages/gamut-labs/babel.config.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+    [
+      'emotion',
+      {
+        sourceMap: true,
+        autoLabel: process.env.NODE_ENV !== 'production',
+        labelFormat: '[local]',
+        cssPropOptimization: true,
+      },
+    ],
+  ],
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/babel.config.js:14]: Should we DRY this in some way? Re-export the core babel config?

File: packages/gamut/babel.config.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
-  plugins: ['react-anonymous-display-name'],
+  plugins: [
+    'react-anonymous-display-name',
+    [
+      'emotion',
+      {
+        sourceMap: true,
+        autoLabel: process.env.NODE_ENV !== 'production',
+        labelFormat: '[local]',
+        cssPropOptimization: true,
```
Comment (JoshuaKGoldberg) [inline packages/gamut/babel.config.js:11]: Non-blocking tangent: I can't find any good docs on this; https://emotion.sh/docs/babel-plugin-emotion is pretty sparse. Do you have anything?

Comment (codecaaron) [reply packages/gamut/babel.config.js:11]: Their repo markdown is identical :*( 

https://github.com/emotion-js/emotion/tree/master/packages/babel-plugin-emotion

Not a huge amount in there if we want to poke around.

File: packages/gamut-labs/babel.config.js (hunk 1 above)
Comment (codecaaron) [reply packages/gamut-labs/babel.config.js:14]: Yeah I was thinking we'd migrate: https://github.com/Codecademy/babel-preset-codecademy to client-modules and add it there?

**Pull request merged**